no-literal-checker/src/main/java/org/checkerframework/checker/noliteral/NoLiteralAnnotatedTypeFactory.java

```diff
@@ -0,0 +1,191 @@
+package org.checkerframework.checker.noliteral;
+
+import java.util.List;
+import javax.lang.model.element.AnnotationMirror;
+import javax.lang.model.element.Element;
+import javax.lang.model.type.TypeKind;
+import javax.lang.model.type.TypeMirror;
+import javax.lang.model.util.Elements;
+import org.checkerframework.checker.noliteral.qual.MaybeDerivedFromConstant;
+import org.checkerframework.checker.noliteral.qual.NonConstant;
+import org.checkerframework.common.basetype.BaseAnnotatedTypeFactory;
+import org.checkerframework.common.basetype.BaseTypeChecker;
+import org.checkerframework.framework.qual.TypeUseLocation;
+import org.checkerframework.framework.type.AnnotatedTypeFactory;
+import org.checkerframework.framework.type.AnnotatedTypeMirror;
+import org.checkerframework.framework.type.AnnotatedTypeMirror.AnnotatedArrayType;
+import org.checkerframework.framework.type.AnnotatedTypeMirror.AnnotatedDeclaredType;
+import org.checkerframework.framework.type.typeannotator.ListTypeAnnotator;
+import org.checkerframework.framework.type.typeannotator.TypeAnnotator;
+import org.checkerframework.framework.util.defaults.QualifierDefaults;
+import org.checkerframework.javacutil.AnnotationBuilder;
+import org.checkerframework.javacutil.ElementUtils;
+import org.checkerframework.javacutil.TypesUtils;
+
+/** The type factory for the no literal checker. */
+public class NoLiteralAnnotatedTypeFactory extends BaseAnnotatedTypeFactory {
+
+  /** The canonical maybe-constant annotation */
+  private final AnnotationMirror MAYBE_CONSTANT =
+      AnnotationBuilder.fromClass(elements, MaybeDerivedFromConstant.class);
+
+  /** The canonical non-constant annotation */
+  private final AnnotationMirror NON_CONSTANT =
+      AnnotationBuilder.fromClass(elements, NonConstant.class);
+
+  public NoLiteralAnnotatedTypeFactory(BaseTypeChecker checker) {
+    super(checker);
+    this.postInit();
+  }
+
+  /** @return a canonical non-constant AnnotationMirror */
+  public AnnotationMirror getNonConstant() {
+    return NON_CONSTANT;
+  }
+
+  /** @return a canonical maybe-constant AnnotationMirror */
+  public AnnotationMirror getMaybeConstant() {
+    return MAYBE_CONSTANT;
+  }
+
+  @Override
+  protected TypeAnnotator createTypeAnnotator() {
+    return new ListTypeAnnotator(new NoLiteralTypeAnnotator(this), super.createTypeAnnotator());
+  }
+
+  @Override
+  protected final QualifierDefaults createQualifierDefaults() {
+    QualifierDefaults defaults = new NoLiteralQualifierDefaults(elements, this);
+    // Ensure that unchecked code is treated optimistically by switching the
+    // standard defaults for unchecked code.
+    defaults.addUncheckedCodeDefault(MAYBE_CONSTANT, TypeUseLocation.PARAMETER);
+    defaults.addUncheckedCodeDefault(NON_CONSTANT, TypeUseLocation.RETURN);
```

Comment:
I can (and will) use `STANDARD_UNCHECKED_DEFAULTS_TOP`, which I did not notice before. However, I cannot use `STANDARD_UNCHECKED_DEFAULTS_BOTTOM`, because I don't want to change the default for lower bounds, only for parameters (because of the other changes I make to defaulting for type variables later). I'll add a comment.